ref-exchange/src/stable_swap/mod.rs

```diff
@@ -0,0 +1,1084 @@
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::collections::LookupMap;
+use near_sdk::json_types::ValidAccountId;
+use near_sdk::{env, AccountId, Balance, Timestamp};
+
+use crate::admin_fee::AdminFees;
+use crate::errors::*;
+use crate::stable_swap::math::{
+    Fees, StableSwap, SwapResult, MAX_AMP, MAX_AMP_CHANGE, MIN_AMP, MIN_RAMP_DURATION,
+};
+use crate::utils::{add_to_collection, SwapVolume, FEE_DIVISOR, U256};
+use crate::StorageKey;
+
+mod math;
+
+pub const MIN_DECIMAL: u8 = 1;
+pub const MAX_DECIMAL: u8 = 18;
+pub const TARGET_DECIMAL: u8 = 18;
```

Comment:
why is it 18? on NEAR we use 24 - so I would expect some native stable coins potentially to use 24.

Reply:
There are some concerns about that. First of all, it is too much using 24 decimals as a stable coin. As far as I know the maximum decimals for a stable coin is DAI in ETH, it is 18. Second, although NEAR use 24 decimals for its origin coin, most of native token on NEAR native chain use 18 decimals, including PARAS, PULSE, SKYWARD, OCT and etc. Last, 18 decimals is acceptable to users from other chains especially ETH ecosystem.  
So, I would suggest to use 18 instead 24 for our stable swap pool LP token, as we wish it is not only a simple lp share but more a token to circulate outside.